fe/fe-core/src/main/java/org/apache/doris/nereids/rules/mv/SelectRollup.java

```diff
@@ -0,0 +1,410 @@
+        Set<String> nonEqualColNames = split.getOrDefault(false, ImmutableSet.of());
+
+        // 2. find matching key prefix most.
+        List<MaterializedIndex> matchingKeyPrefixMost;
+        if (!(equalColNames.isEmpty() && nonEqualColNames.isEmpty())) {
+            List<MaterializedIndex> matchingResult = matchKeyPrefixMost(table, containAllRequiredColumns,
+                    equalColNames, nonEqualColNames);
+            matchingKeyPrefixMost = matchingResult.isEmpty() ? containAllRequiredColumns : matchingResult;
+        } else {
+            matchingKeyPrefixMost = containAllRequiredColumns;
+        }
+
+        List<Long> partitionIds = olapScan.getSelectedPartitionIds();
+        // 3. sort by row count, column count and index id.
+        return matchingKeyPrefixMost.stream()
+                .map(MaterializedIndex::getId)
+                .sorted(Comparator
+                        // compare by row count
+                        .comparing(rid -> partitionIds.stream()
+                                .mapToLong(pid -> table.getPartition(pid).getIndex((Long) rid).getRowCount())
+                                .sum())
+                        // compare by column count
+                        .thenComparing(rid -> table.getSchemaByIndexId((Long) rid).size())
+                        // compare by rollup index id
+                        .thenComparing(rid -> (Long) rid))
+                .collect(Collectors.toList());
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Matching key prefix
+    ///////////////////////////////////////////////////////////////////////////
+    private List<MaterializedIndex> matchKeyPrefixMost(
+            OlapTable table,
+            List<MaterializedIndex> rollups,
+            Set<String> equalColumns,
+            Set<String> nonEqualColumns) {
+        TreeMap<Integer, List<MaterializedIndex>> collect = rollups.stream()
+                .collect(Collectors.toMap(
+                        rollup -> rollupKeyPrefixMatchCount(table, rollup, equalColumns, nonEqualColumns),
+                        Lists::newArrayList,
+                        (l1, l2) -> {
+                            l1.addAll(l2);
+                            return l1;
+                        },
+                        TreeMap::new)
+                );
+        return collect.descendingMap().firstEntry().getValue();
+    }
+
+    private int rollupKeyPrefixMatchCount(
+            OlapTable table,
+            MaterializedIndex rollup,
+            Set<String> equalColNames,
+            Set<String> nonEqualColNames) {
+        int matchCount = 0;
+        for (Column column : table.getSchemaByIndexId(rollup.getId())) {
+            if (equalColNames.contains(column.getName())) {
+                matchCount++;
+            } else if (nonEqualColNames.contains(column.getName())) {
+                // Unequivalence predicate's columns can match only first column in rollup.
```

Comment:
I know this logic is same as the origin selector, but would you like to explain why `Unequivalence predicate's columns can match only first column in rollup.`? thx